packages/editor/src/lib/exports/exportToSvg.tsx

```diff
@@ -0,0 +1,128 @@
+import { TLShapeId } from '@tldraw/tlschema'
+import { assert } from '@tldraw/utils'
+import { flushSync } from 'react-dom'
+import { createRoot } from 'react-dom/client'
+import { Editor } from '../editor/Editor'
+import { TLImageExportOptions } from '../editor/types/misc-types'
+import { StyleEmbedder } from './StyleEmbedder'
+import { embedMedia } from './embedMedia'
+import { getSvgJsx } from './getSvgJsx'
+
+export async function exportToSvg(
+	editor: Editor,
+	shapeIds: TLShapeId[],
+	opts: TLImageExportOptions = {}
+) {
+	// when rendering to SVG, we start by creating a JSX representation of the SVG that we can
+	// render with react. Hopefully elements will have a `toSvg` method that renders them to SVG,
+	// but if they don't we'll render their normal HTML content into an svg <foreignObject> element.
+	const result = getSvgJsx(editor, shapeIds, opts)
+	if (!result) return undefined
+
+	// we need to render that SVG into a real DOM element that's actually laid out in the document.
+	// without this CSS and layout aren't computed correctly, which we need to make sure any
+	// <foreignObject> elements have their styles and content inlined correctly.
+	const container = editor.getContainer()
+	const renderTarget = document.createElement('div')
+	renderTarget.className = 'tldraw-svg-export'
+	renderTarget.inert = true
```

Comment:
this is slightly different than the above - you have `inert` but also a display: none but not the opacity: 0. and also this one has the tabIndex but the other one doesn't.
any reason they should be different? should they be consistent?

Reply:
sorry, the above where?

Reply:
oh i see!

yea so for this one, we need full layout etc. which is why this goes in the document and is rendered at 0 opacity.

for the other one, we need `contentDocument`, which is only populated once the iframe is added to the document. we don't need full layout tho, which is why we can use `display: none`. we don't need `inert` there either because of that, so i've removed it

Reply:
gotcha gotcha - ok, makes sense!